purchase_order_supplierinfo_update/tests/test_purchase_order_supplierinfo_update.py

```diff
@@ -0,0 +1,132 @@
+        po_form_1.partner_id = self.supplier
+        now = fields.Datetime.now()
+        po_form_1.date_order = now - relativedelta(days=1)
+        with po_form_1.order_line.new() as po_line_form:
+            po_line_form.product_id = self.product
+            po_line_form.taxes_id.clear()
+        purchase_order_1 = po_form_1.save()
+        purchase_order_1.button_confirm()
+        # Create second purchase
+        po_form_2 = Form(self.env["purchase.order"])
+        po_form_2.partner_id = self.supplier
+        with po_form_2.order_line.new() as po_line_form:
+            po_line_form.product_id = self.product
+            po_line_form.price_unit = 200
+            po_line_form.taxes_id.clear()
+        purchase_order_2 = po_form_2.save()
+        purchase_order_2.button_confirm()
+        # Change price in second purchase
+        with Form(purchase_order_2) as po_form_2:
+            with po_form_2.order_line.edit(0) as po_line_form:
+                po_line_form.price_unit = 300
+        self.assertEqual(self.supplierinfo.price, 300)
+        # Change price in first purchase. This doesn't update supplierinfo
+        # because it isn't the last purchase of this product by this supplier
+        with Form(purchase_order_1) as po_form_1:
+            with po_form_1.order_line.edit(0) as po_line_form:
+                po_line_form.price_unit = 400
+        self.assertEqual(self.supplierinfo.price, 300)
+
+    def test_create_new_line_in_a_confirmed_po(self):
+        # Create first purchase
+        po_form_1 = Form(self.env["purchase.order"])
+        po_form_1.partner_id = self.supplier
+        with po_form_1.order_line.new() as po_line_form:
+            po_line_form.product_id = self.product
+            po_line_form.taxes_id.clear()
+        purchase_order_1 = po_form_1.save()
+        purchase_order_1.button_confirm()
+        # Create second purchase
+        po_form_2 = Form(self.env["purchase.order"])
+        po_form_2.partner_id = self.supplier
+        with po_form_2.order_line.new() as po_line_form:
+            po_line_form.product_id = self.product
+            po_line_form.price_unit = 200
+            po_line_form.taxes_id.clear()
+        purchase_order_2 = po_form_2.save()
+        purchase_order_2.button_confirm()
+        # Create a new line in the first purchase (that is already confirmed)
+        # with another product.
+        # We can not use a Form due to a modifier restriction on purchase order view
+        # purchase/views/purchase_views.xml#L230
+        purchase_order_1.write(
+            {
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "product_id": self.product_2.id,
+                            "product_uom": self.product_2.uom_po_id.id,
+                            "price_unit": 20.00,
+                        },
+                    )
+                ]
```

Comment:
Coud you use `Command.create` It is more readable. [Example here ](https://github.com/OCA/account-financial-tools/blob/d1d31e020860fc3ddf93fce49944a17b28bf4754/account_move_name_sequence/tests/test_account_move_name_seq.py#L84)
```suggestion
                "order_line": [
                    Command.create({
                            "product_id": self.product_2.id,
                            "product_uom": self.product_2.uom_po_id.id,
                            "price_unit": 20.00,
                        })
                ]
```

Reply:
ping @jdidderen-nsi